daemon/daemon.c

```diff
@@ -58,6 +58,23 @@ void create_needed_dir(const char *dir, uid_t uid, gid_t gid)
         error("Cannot create directory '%s'", dir);
 }
 
+void clean_directory(char *dirname)
+{
+    DIR *dir = opendir(dirname);
+    if(!dir) return;
+
+    struct dirent *de = NULL;
+
+    while((de = readdir(dir)))
+        if(de->d_type == DT_REG) {
+            char filename[FILENAME_MAX +1];
+            snprintfz(filename, FILENAME_MAX, "%s/%s", dirname, de->d_name);
+            unlink(filename);
```

Comment:
Please, test `unlink` return, and give an additional error message to the log.

Reply:
Done.